src/contract/mod.rs

```diff
@@ -105,37 +105,35 @@ impl<T: Transport> Contract<T> {
         self.address
     }
 
-    /// Execute a contract function
-    pub fn call<P>(&self, func: &str, params: P, from: Address, options: Options) -> CallFuture<H256, T::Out>
+    /// Returns calldata for contract function call
+    pub fn construct_calldata<P>(&self, func: &str, params: P) -> Vec<u8>
     where
         P: Tokenize,
     {
         self.abi
             .function(func)
             .and_then(|function| function.encode_input(&params.into_tokens()))
-            .map(move |data| {
-                let Options {
-                    gas,
-                    gas_price,
-                    value,
-                    nonce,
-                    condition,
-                } = options;
+            .unwrap()
```

Comment:
The error is there for a reason, simply `unwrap`ing will cause panics in case the function is not found.

Reply:
Thanks for pointing me towards [#17](https://github.com/tomusdrw/rust-web3/issues/17) .. ethabi_derive looks really nice :ok_hand: 
Sorry about the unwrap() .. should have been an issue not a pull request :roll_eyes:

Reply:
No worries @kyledewy, thanks a lot for the contribution!